.github/workflows/cd.yml

```diff
@@ -0,0 +1,154 @@
+---
+name: CD
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - "src/**"
+  workflow_dispatch:
+    inputs:
+      release_type:
+        description: 'Release type'
+        required: true
+        default: 'Dry Run'
+        options:
+          - 'Dry Run'
+          - 'Release'
+
+jobs:
+  version-type:
+    uses: ./.github/workflows/_version_type.yml
+
+  version-bump:
+    name: Version bump
+    runs-on: ubuntu-22.04
+    needs: version-type
+    outputs:
+      version: ${{ steps.get-version.outputs.version }}
+    steps:
+      - name: Check out repo
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+
+      - name: Set up Python
+        uses: actions/setup-python@0a5c61591373683505ea898e09a3ea4f39ef2b9c # v5.0.0
+        with:
+          python-version-file: ".python-version"
+
+      - name: Install hatch
+        run: pip install hatch
+
+      - name: Get current version
+        run: echo "OLD_VERSION=$(hatch version)" >> $GITHUB_ENV
+
+      - name: Bump version
+        run: hatch bump ${{ needs.version-type.outputs.type }}
```

Comment:
⚠️ This looks like it's calling the incorrect output, by the looks of `_version_type`.
```yaml
on:
  workflow_call:
    outputs:
      version:
        description: "version to be built"
        value: ${{ jobs.version.outputs.bump_type }}
```

it should be:
```suggestion
        run: hatch bump ${{ needs.version-type.outputs.version }}
```